pkg/cluster/mpi/statefulset.go

```diff
@@ -33,11 +33,17 @@ var (
 // Key of the shared Secret object that contains client-side SSH public key
 const publicKeyField = "ssh-publickey"
 
-func StatefulSet() core.OwnedComponent {
-	return &statefulSetComponent{}
+func StatefulSet(initImage, syncImage string) core.OwnedComponent {
```

Comment:
Hmm, something about this feels off to me - too specific to our implementation and therefore leaking out the encapsulation. Should this be a list of init images and a list of sidecar images rather than one of each, with a very specific name in the second case?

Reply:
No quite... 
Here's the idea: All DCO cluster types uniformly allow to specify in CRD a list of init containers that may have some significance to the user. (There is no way to specify side container in a similar manner). Regardless of that, for MPI we want to have one init and one sidecar container that serve special purposes. The DCO controller appends those containers to whatever is received through CRD. The two parameters in question are used to override defaults of those images based on controller's startup arguments set by the administrator through a helm chart -- and this is in addition to what Nucleus may request through CRD.